pkg/kds/zone/components_test.go

```diff
@@ -94,7 +88,9 @@ var _ = Describe("Zone Sync", func() {
 		zoneStore = memory.NewStore()
 		zoneSyncer = sync_store.NewResourceSyncer(core.Log.WithName("kds-syncer"), zoneStore)
 
-		start(newPolicySink(zoneName, zoneSyncer, clientStream, &testRuntimeContext{kds: kdsCtx}), stop)
+		go func() {
+			_ = newPolicySink(zoneName, zoneSyncer, clientStream, &testRuntimeContext{kds: kdsCtx}).Start()
+		}()
```

Comment:
I get a data race detected here:
```
WARNING: DATA RACE
Write at 0x00c00098ef70 by goroutine 62:
  github.com/kumahq/kuma/pkg/kds/zone_test.glob..func1.3()
      /home/jpeach/upstream/konghq/kuma/pkg/kds/zone/components_test.go:89 +0x679
...

Previous read at 0x00c00098ef70 by goroutine 140:
  github.com/kumahq/kuma/pkg/kds/zone_test.glob..func1.3.1()
      /home/jpeach/upstream/konghq/kuma/pkg/kds/zone/components_test.go:92 +0xbb

```

Reply:
Yes that's part of the things I still need to check but I think it's a test only issue.